### PR TITLE
[search] Fixed search in viewport filtering.

### PR DESCRIPTION
**Pull request by ygorshenin**
https://jira.mail.ru/browse/MAPSME-2446

@mpimenov PTAL

### CODE REVIEW

File: search/ranker.cpp [hunk 1]
```diff
@@ -509,8 +510,8 @@ void Ranker::UpdateResults(bool lastUpdate)
 
   // Emit feature results.
   size_t count = m_emitter.GetResults().GetCount();
-  size_t i;
-  for (i = 0; i < m_tentativeResults.size(); ++i)
+  size_t i = 0;
+  for (; i < m_tentativeResults.size(); ++i)
   {
     if (!lastUpdate && i >= kBatchSize && !m_params.m_viewportSearch)
       break;
```
Comment (kshalnev) [inline search/ranker.cpp]: [optional, may be other PR]. Next line is calling a function 'BailIfCancelled' where 'Bail' is looks like a misprint of 'Fail'. May be then rename BailIfCancelled to FailIfCancelled?

File: search/intermediate_result.cpp [hunk 2]
```diff
@@ -24,7 +24,6 @@ namespace search
 {
 
 /// All constants in meters.
```
Comment (kshalnev) [inline search/intermediate_result.cpp]: Now it is only one constant in meters. Then, may be write this comment on right side of declarations? Like this:
double const DIST_SAME_STREET = 5000.0; // in meters

File: search/ranker.cpp (hunk 1 above)
Comment (ygorshenin) [inline search/ranker.cpp]: RLY? http://www.urbandictionary.com/define.php?term=bail

File: search/intermediate_result.cpp (hunk 2 above)
Comment (ygorshenin) [inline search/intermediate_result.cpp]: What's the profit?

Comment (ygorshenin) [inline search/intermediate_result.cpp]: It would be much better to rename it to DIST_SAME_STREET_METERS

File: search/search_params.hpp [hunk 3]
```diff
@@ -49,7 +49,7 @@ class SearchParams
 
   double m_lat, m_lon;
 
-  // A minimum distance between search results in mercator, needed for
+  // A minimum distance between search results in meters, needed for
   // pre-ranking of viewport search results.
   double m_minDistanceOnMapBetweenResults;
```
Comment (kshalnev) [inline search/search_params.hpp]: [optional] This struct looks a bit irregularly - some members are accessible via set/get pairs, while other are accessible directly. May be do it in a one manner? It is not about the PR, just as concomitant note.

File: search/ranker.cpp (hunk 1 above)
Comment (kshalnev) [inline search/ranker.cpp]: ok, thanks.

File: search/intermediate_result.cpp (hunk 2 above)
Comment (kshalnev) [inline search/intermediate_result.cpp]: profit - code/comment would be a bit more accurate. xxx_METERS is even better.

Comment (ygorshenin) [inline search/intermediate_result.cpp]: It's quite arguable what's better - a comment on an individual line or an inline comment.

Comment (kshalnev) [inline search/intermediate_result.cpp]: my note was not about comments style - like a group or for each const. it was about counting - comment says about many constants, while there is only one constant.
it is not critical. up to you.

Comment (ygorshenin) [inline search/intermediate_result.cpp]: Comment said "All constants", not "All constants, where number of constants is greater than one."

**Comment by ygorshenin** [comment]
PTAL

File: search/search_params.hpp (hunk 3 above)
Comment (ygorshenin) [inline search/search_params.hpp]: Done

**Comment by kshalnev** [comment]
LGTM

**Comment by ygorshenin** [comment]
@mpimenov PTAL

File: search/intermediate_result.cpp [hunk 4]
```diff
@@ -277,9 +275,8 @@ bool PreResult2::EqualLinearTypesF::operator() (PreResult2 const & r1, PreResult
 {
   // Note! Do compare for distance when filtering linear objects.
   // Otherwise we will skip the results for different parts of the map.
-  return (r1.m_geomType == feature::GEOM_LINE &&
-          r1.IsEqualCommon(r2) &&
-          PointDistance(r1.GetCenter(), r2.GetCenter()) < DIST_SAME_STREET);
+  return (r1.m_geomType == feature::GEOM_LINE && r1.IsEqualCommon(r2) &&
```
Comment (mpimenov) [inline search/intermediate_result.cpp]: Feel free to remove the parentheses.

File: search/processor.cpp [hunk 5]
```diff
@@ -110,8 +110,9 @@ void SendStatistics(SearchParams const & params, m2::RectD const & viewport, Res
   string posX, posY;
   if (params.IsValidPosition())
   {
-    posX = strings::to_string(MercatorBounds::LonToX(params.m_lon));
-    posY = strings::to_string(MercatorBounds::LatToY(params.m_lat));
+    m2::PointD const position = params.GetPositionMercator();
```
Comment (mpimenov) [inline search/processor.cpp]: ```auto const pos = params.GetPositionMercator();```?

File: search/intermediate_result.cpp (hunk 4 above)
Comment (ygorshenin) [inline search/intermediate_result.cpp]: Please, take a look at the CL title - "Fixed search in viewport filtering", not "... filtering and whole legacy code refactoring".

**Comment by ygorshenin** [comment]
@mpimenov PTAL

**Comment by mpimenov** [comment]
LGTM

**Pull request merged**